cmd/disk-cache-backend.go

```diff
@@ -522,12 +598,61 @@ func (c *diskCache) bitrotReadFromCache(ctx context.Context, filePath string, of
 	return nil
 }
 
+// decryptCacheObjectETag tries to decrypt the ETag saved in encrypted format using the cache KMS
+func decryptCacheObjectETag(info *ObjectInfo) (error) {
+	// Directories are never encrypted.
+	if info.IsDir {
+		return nil
+	}
+	encrypted := crypto.S3.IsEncrypted(info.UserDefined) && isCacheEncrypted(info.UserDefined)
+
+	switch {
+	case encrypted:
+		if globalCacheKMS == nil {
+			return errKMSNotConfigured
+		}
+		keyID, kmsKey, sealedKey, err := crypto.S3.ParseMetadata(info.UserDefined)
+
+		if err != nil {
+			return err
+		}
+		extKey, err := globalCacheKMS.UnsealKey(keyID, kmsKey, crypto.Context{info.Bucket: path.Join(info.Bucket, info.Name)})
+		if err != nil {
+			return err
+		}
+		var objectKey crypto.ObjectKey
+		if err = objectKey.Unseal(extKey, sealedKey, crypto.S3.String(), info.Bucket, info.Name); err != nil {
+			return err
+		}
+		info.ETag = tryDecryptCacheETag(objectKey[:], info.ETag, false)
+		return nil
+	}
+
+	return nil
+}
+
+// helper to decrypt Etag given object encryption key and encrypted ETag
+func tryDecryptCacheETag(key []byte, encryptedETag string, ssec bool) string {
```

Comment:
Isn't that the same as `tryDecryptETag`?

Reply:
yes, will reuse the tryDecryptETag here.